python/ray/tune/trainable.py

```diff
@@ -442,11 +442,18 @@ def save(self, checkpoint_dir=None) -> str:
             checkpoint, parent_dir=checkpoint_dir, trainable_state=trainable_state
         )
 
+        self.postprocess_checkpoint(checkpoint_path)
+
         # Maybe sync to cloud
         self._maybe_save_to_cloud(checkpoint_dir)
 
         return checkpoint_path
 
+    def postprocess_checkpoint(self, checkpoint_path: str):
+        """Run any additional postprocessing before the checkpoint
+        is saved to cloud."""
+        pass
+
```

Comment:
Let's make this an internal API for now (then we can change the implementation later)